feg/cloud/go/services/feg_relay/gw_to_feg_relay/servicers/session_proxy_relay.go

```diff
@@ -71,33 +72,56 @@ func (s *RelayRouter) UpdateSession(
 			req.UsageMonitors = append(req.UsageMonitors, m)
 		}
 	}
-	resultChan := make(chan *protos.UpdateSessionResponse, len(reqMap))
+
+	/*Each request in reqMap is segregated into two requests and relays each request to respective proxies (session_proxy and n7n40_proxy).
+	Hence length of resultChan is len(reqMap) * 2
+	*/
+	resultChan := make(chan *protos.UpdateSessionResponse, len(reqMap)*2)
+	var sessionProxyReq *protos.UpdateSessionRequest
+	var n7n40ProxyReq *protos.UpdateSessionRequest
+	proxyMap := make(map[gateway_registry.GwServiceType]*protos.UpdateSessionRequest)
 
 	// send a separate Update request for each unique PLMN ID
 	for plmnid, req := range reqMap {
-		go func(plmnid string, req *protos.UpdateSessionRequest) {
-			client, ctx, cancel, err := s.getSessionProxyClient(ctx, plmnid)
-			if err != nil {
-				glog.Errorf("failed connect to Session Proxy for PLMNID '%s': %v", plmnid, err)
-				resultChan <- genUpdateErrorResp(req)
-				return
-			}
-			defer cancel()
-			resp, err := client.UpdateSession(ctx, req)
-			if err != nil {
-				glog.Errorf("failed Session Proxy Update for PLMNID '%s': %v", plmnid, err)
-				resultChan <- genUpdateErrorResp(req)
-				return
+		for i := 0; i < len(req.Updates); i++ {
+			if req.Updates[i].CommonContext.RatType == protos.RATType_TGPP_NR {
+				n7n40ProxyReq.Updates = append(n7n40ProxyReq.Updates, req.Updates[i])
+				n7n40ProxyReq.UsageMonitors = append(n7n40ProxyReq.UsageMonitors, req.UsageMonitors[i])
+			} else {
+				sessionProxyReq.Updates = append(sessionProxyReq.Updates, req.Updates[i])
+				sessionProxyReq.UsageMonitors = append(sessionProxyReq.UsageMonitors, req.UsageMonitors[i])
```

Comment:
How is it going to work???

sessionProxyReq AND n7n40ProxyReq are both nil at this point. You are going to crash this service as soon as you reference a field of any of them.

How was this PR tested?

Reply:
Corrected. Verified with UT